### PR TITLE
Fixes in Alpine gifv and create Bullseye version

### PR DESCRIPTION
**Pull request by altendorfme**
I created a new PR, the first one had problems!

### Alpine:

I saw that in the Dockerfile that has ffmpeg-dev, but when I enable GIFV I get the warning:

```
thumbor:WARNING gifv optimizer enabled but binary FFMPEG_PATH does not exist
```

I edited the Dockerfile and added ffmpeg, I did a new build and it worked, now we have gifs being converted to MP4 and WebM. 

During build the packages `lapack-dev` and `libressl-dev` gave error:

```
1.860 ERROR: unable to select packages:
1.867   lapack-3.11-r2:
1.867     breaks: liblapack-0.3.23-r0[!lapack]
1.868     satisfies: lapack-dev-3.11-r2[lapack=3.11-r2]
1.871   .build-deps-20230731.204301:
1.871     masked in: cache
1.872     satisfies: world[.build-deps=20230731.204301]
```
```
50.27 ERROR: libressl-dev-3.7.3-r0: trying to overwrite usr/include/openssl/aes.h owned by openssl-dev-3.1.1-r3
```

I update alpine to 10.6 this solves the problem
Alpine has alot os problems to compile using [opencv] tag, openblas doesn't work properly with numpy

### Bullseye:

In the search for corrections I found the article: [Using Alpine can make Python Docker builds 50× slower](https://pythonspeed.com/articles/alpine-docker-python/), I'm not an expert to debate here, but Thumbor's recommendation is to use debian to have a faster and more efficient compilation, with support for gifv and opencv, including the official image that follows this pattern.

I added support for the Bullseye version, and it really works great, I kept all the original design patterns and made a script to test all the filters: [thumbor.pdf](https://github.com/mvhirsch/thumbor-docker/files/12235580/thumbor.pdf)

### CODE REVIEW

**Comment by mvhirsch** [comment]
Thank you very much for your contribution!

That's correct. I didn't install `ffmpeg` on purpose (to keep alpine image small). The fact I temporarily install `ffmpeg-dev` at build time is just for other build dependencies needed. The overall goal of the `alpine` image was to be small, not to be full fledged. Instead that's the `slim` variant is for.

But after upgrading to thumbor 7, the `slim` variant was smaller than the `alpine` variant. That's why I'm thinking about getting rid of it. I can't invest time to make it smaller (again) atm. If you want to invest some time in optimizing the alpine image to be smaller, you're welcome! I would even accept the `ffmpeg` installment (making the image size bigger, though).

About the "bullseyes" version: I don't see any benefit in providing such a variant. Also: `python:3.10.12` [is bookworm](https://hub.docker.com/_/python), not bullseye. While I appreciate updating the base image, I don't see much benefit in maintaining to debian versions here.

> In the search for corrections I found the article: [Using Alpine can make Python Docker builds 50× slower](https://pythonspeed.com/articles/alpine-docker-python/), I'm not an expert to debate here, but Thumbor's recommendation is to use debian to have a faster and more efficient compilation, [...]

That's why I'm using the official `python` image here. They're based on debian. Next: unless you build an alpine image 500 times per day, I don't think the build duration is worthy to think of. Please proof me wrong.

**Comment by mvhirsch** [comment]
> I added support for the Bullseye version, and it really works great, I kept all the original design patterns and made a script to test all the filters: [thumbor.pdf](https://github.com/mvhirsch/thumbor-docker/files/12235580/thumbor.pdf)

I'm interested in that script you made, too. That would help testing new thumbor versions faster. Would you like to provide an PR for that?

**Comment by altendorfme** [comment]
> That's correct. I didn't install `ffmpeg` on purpose (to keep alpine image small). The fact I temporarily install `ffmpeg-dev` at build time is just for other build dependencies needed. The overall goal of the `alpine` image was to be small, not to be full fledged. Instead that's the `slim` variant is for.
> 
> But after upgrading to thumbor 7, the `slim` variant was smaller than the `alpine` variant. That's why I'm thinking about getting rid of it. I can't invest time to make it smaller (again) atm. If you want to invest some time in optimizing the alpine image to be smaller, you're welcome! I would even accept the `ffmpeg` installment (making the image size bigger, though).
> 
> About the "bullseyes" version: I don't see any benefit in providing such a variant. Also: `python:3.10.12` [is bookworm](https://hub.docker.com/_/python), not bullseye. While I appreciate updating the base image, I don't see much benefit in maintaining to debian versions here.

Yes, you're right it's a bookworm image. The benefit is providing the full package of filters and encodes working, could be the alternative with `FFMPEG` support, thus keeping the alpine image lighter.

> > In the search for corrections I found the article: [Using Alpine can make Python Docker builds 50× slower](https://pythonspeed.com/articles/alpine-docker-python/), I'm not an expert to debate here, but Thumbor's recommendation is to use debian to have a faster and more efficient compilation, [...]
> 
> That's why I'm using the official `python` image here. They're based on debian. Next: unless you build an alpine image 500 times per day, I don't think the build duration is worthy to think of. Please proof me wrong.

When I added `opencv-python-headless` to the `Alpine` image there is a problem, it takes more than 20 minutes compiling on an 8 cpus machine and the end result still leaves incomplete filters. Searching on Google is something you see a lot, I've tested 3 or 4 versions of Python and the result is always the same. So I mounted the image with `bookworm` which does the same job in seconds.

**Comment by altendorfme** [comment]
> > I added support for the Bullseye version, and it really works great, I kept all the original design patterns and made a script to test all the filters: [thumbor.pdf](https://github.com/mvhirsch/thumbor-docker/files/12235580/thumbor.pdf)
> 
> I'm interested in that script you made, too. That would help testing new thumbor versions faster. Would you like to provide an PR for that?

I created this PHP, not beautiful but efficient haha

```php
<?php

$images = [
	'https://github.com/thumbor/thumbor/raw/master/example.jpg',
	'https://github.com/thumbor/thumbor/raw/master/docs/images/dice_transparent_background.png',
	'https://www.autoracing.com.br/wp-content/uploads/2023/03/F1-leclerc-australia-2023-treinos-sexta.jpg',
	'https://upload.wikimedia.org/wikipedia/commons/thumb/2/22/Turkish_Van_Cat.jpg/546px-Turkish_Van_Cat.jpg',
];
$images_filters = [
	'filters:autojpg()',
	'filters:background_color(f00)',
	'filters:blur(7)',
	'filters:brightness(40)',
	'filters:contrast(40)',
	'filters:convolution(1;2;1;2;4;2;1;2;1,3,true)',
	'filters:cover()',
	'filters:equalize()',
	'filters:extract_focal()',
	'filters:fill(f00)',
	'filters:focal(146x206:279x360)',
	'filters:format(webp)',
	'filters:grayscale()',
	'filters:max_bytes(7500)',
	'filters:noise(40)',
	'filters:proportion(0.5)',
	'filters:quality(40)',
	'filters:rgb(20,-20,40)',
	'filters:rotate(90)',
	'filters:round_corner(20|40,255,255,255)',
	'filters:saturation(40)',
	'filters:sharpen(2,1.0,true)',
	'filters:stretch()',
	'filters:watermark(https://upload.wikimedia.org/wikipedia/commons/3/31/Wiki_logo_Nupedia.jpg,10,10,20,30,30)',
	'smart'
];


$gifs = [
	'https://www.litmus.com/wp-content/uploads/2021/02/ease-applied-to-tween-with-bouncein-example.gif',
	'https://i.imgur.com/1THykVa.gif',
	'https://i.imgur.com/eTTzjIz.gif',
];
$gifs_filters = [
	'filters:stretch()',
	'smart',
	'filters:gifv()',
	'filters:gifv(mp4)',
	'filters:gifv(webm)'
];
?>

<h1>Images</h1>

<?php
foreach ($images_filters as $filter) { ?>

	<h2><?php echo $filter ?></h2>
	<table>
		<?php
		foreach ($images as $image) {
			if ($filter == 'filters:fill(f00)') {
				$size = 'fit-in/200x350';
			} else {
				$size = '200x150';
			}

		?>
			<td>
				<img src="https://test.test/unsafe/<?php echo $size; ?>/<?php echo $filter; ?>/<?php echo $image; ?>" />
			</td>
		<?php }
		?>
	</table>
<?php
} ?>

<h1>Gifs/Videos</h1>

<?php
foreach ($gifs_filters as $filter) { ?>
	<h2><?php echo $filter ?></h2>
	<table>
		<?php
		foreach ($gifs as $gif) {
			if ($filter == 'filters:gifv()' || $filter == 'filters:gifv(mp4)' || $filter == 'filters:gifv(webm)') {
				if (strpos($gif, '.gif') !== false) { ?>
					<td>
						<video width="200" height="200" controls>
							<source src="https://test.test/unsafe/<?php echo $size; ?>/<?php echo $filter; ?>/<?php echo $gif; ?>" type="video/mp4">
						</video>
					</td>
				<?php }
			} else { ?>
				<td>
					<img src="https://test.test/unsafe/<?php echo $size; ?>/<?php echo $filter; ?>/<?php echo $gif; ?>" />
				</td>
		<?php }
		}
		?>
	</table>
<?php
}

```

**Comment by mvhirsch** [comment]
> The benefit is providing the full package of filters and encodes working [...]

Are you talking about the alpine or the slim variant? I'm not sure now. The slim should provide a full package of filters, while the alpine does not. Thus a bullseye image is not needed.

**Comment by mvhirsch** [comment]
> I created this PHP, not beautiful but efficient haha

Thank you very much. Could you please create a PR for that?

**Comment by altendorfme** [comment]
> > The benefit is providing the full package of filters and encodes working [...]
> 
> Are you talking about the alpine or the slim variant? I'm not sure now. The slim should provide a full package of filters, while the alpine does not. Thus a bullseye image is not needed.

Maybe I failed to test the Slim version more carefully, but my tests were by Alpine

**Comment by altendorfme** [comment]
Now I test and remembered, in the `slim` version, when enabling the filters it returns the following error:

```
Traceback (most recent call last):
  File "/usr/local/thumbor/.local/bin/thumbor", line 8, in <module>
    sys.exit(main())
  File "/usr/local/thumbor/.local/lib/python3.10/site-packages/thumbor/server.py", line 158, in main
    importer = get_importer(config)
  File "/usr/local/thumbor/.local/lib/python3.10/site-packages/thumbor/server.py", line 63, in get_importer
    importer.import_modules()
  File "/usr/local/thumbor/.local/lib/python3.10/site-packages/thumbor/importer.py", line 86, in import_modules
    self.import_item("DETECTORS", "Detector", is_multiple=True)
  File "/usr/local/thumbor/.local/lib/python3.10/site-packages/thumbor/importer.py", line 135, in import_item
    self.load_multiple_item(
  File "/usr/local/thumbor/.local/lib/python3.10/site-packages/thumbor/importer.py", line 167, in load_multiple_item
    module = self.import_class(
  File "/usr/local/thumbor/.local/lib/python3.10/site-packages/thumbor/importer.py", line 56, in import_class
    kls = import_class(name, get_module)
  File "/usr/local/thumbor/.local/lib/python3.10/site-packages/thumbor/importer.py", line 20, in import_class
    module = import_module(module_name)
  File "/usr/local/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/usr/local/thumbor/.local/lib/python3.10/site-packages/thumbor/detectors/face_detector/__init__.py", line 13, in <module>
    from thumbor.detectors.local_detector import CascadeLoaderDetector
  File "/usr/local/thumbor/.local/lib/python3.10/site-packages/thumbor/detectors/local_detector.py", line 14, in <module>
    import numpy as np
ModuleNotFoundError: No module named 'numpy'

```

File: tests.php [hunk 1]
```diff
@@ -0,0 +1,102 @@
+<?php
+
+$images = [
```
Comment (mvhirsch) [inline tests.php:3]: @altendorfme please create a new PR with this file.

**Comment by mvhirsch** [comment]
> > > The benefit is providing the full package of filters and encodes working [...]
> > 
> > 
> > Are you talking about the alpine or the slim variant? I'm not sure now. The slim should provide a full package of filters, while the alpine does not. Thus a bullseye image is not needed.
> 
> Maybe I failed to test the Slim version more carefully, but my tests were by Alpine

Would you like to create a new PR adding ffmpeg to the alpine variant?

> ModuleNotFoundError: No module named 'numpy'

Looks like it's missing numpy. Could you please create a new PR fixing this in slim version?

**Pull request closed without merging**